<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>830</width>
    <height>656</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>C64++</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="mainScreenFrame">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>30</y>
      <width>831</width>
      <height>541</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>0</number>
     </property>
    </layout>
   </widget>
   <widget class="QPushButton" name="pause">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>pause/continue</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="reset">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>reset</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="keyboard">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>keyboard</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="joystick">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>joystick</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="mute">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>toggle mute</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QSlider" name="volume">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>0</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="step">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>step single cycle</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="step_instruction">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>step instruction</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="stepline">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>step line</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="stepframe">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>step frame</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="breakpoints">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>breakpoints</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="mpu_viewer">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>MPU viewer</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="ram_viewer">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>RAM viewer</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="cia_viewer">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>CIA viewer</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="vic_viewer">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>0</y>
      <width>21</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>VIC viewer</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>830</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_PRG"/>
    <addaction name="actionEnter_Hex_Data"/>
    <addaction name="actionExtract_RAM_to_File"/>
   </widget>
   <widget class="QMenu" name="menuEmulation">
    <property name="title">
     <string>Emulation</string>
    </property>
    <addaction name="actionPause"/>
    <addaction name="actionStep"/>
    <addaction name="actionStep_Instruction"/>
    <addaction name="actionStep_Line"/>
    <addaction name="actionStep_Frame"/>
    <addaction name="separator"/>
    <addaction name="actionHard_Reset"/>
   </widget>
   <widget class="QMenu" name="menuDebug">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionMPU_Viewer"/>
    <addaction name="actionRAM_Viewer"/>
    <addaction name="actionCIA_Viewer"/>
    <addaction name="actionVIC_Viewer"/>
    <addaction name="separator"/>
    <addaction name="actionVirtual_Keyboard"/>
    <addaction name="actionVirtual_Joysticks"/>
    <addaction name="separator"/>
    <addaction name="actionBreakpoint_Editor"/>
   </widget>
   <widget class="QMenu" name="menuInput">
    <property name="title">
     <string>Input</string>
    </property>
    <addaction name="actiondisable_joysticks"/>
    <addaction name="actionjoystick_1"/>
    <addaction name="actionjoystick_2"/>
    <addaction name="actionboth_joysticks"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEmulation"/>
   <addaction name="menuInput"/>
   <addaction name="menuDebug"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionStep">
   <property name="text">
    <string>Step</string>
   </property>
   <property name="shortcut">
    <string>Alt+S</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="text">
    <string>Pause</string>
   </property>
   <property name="shortcut">
    <string>Alt+P</string>
   </property>
  </action>
  <action name="actionHard_Reset">
   <property name="text">
    <string>Hard Reset</string>
   </property>
   <property name="shortcut">
    <string>Alt+H</string>
   </property>
  </action>
  <action name="actionStep_Frame">
   <property name="text">
    <string>Step Frame</string>
   </property>
   <property name="shortcut">
    <string>Alt+F</string>
   </property>
  </action>
  <action name="actionMPU_Viewer">
   <property name="text">
    <string>MPU Viewer</string>
   </property>
   <property name="shortcut">
    <string>Alt+M</string>
   </property>
  </action>
  <action name="actionEnter_Hex_Data">
   <property name="text">
    <string>Enter Hex Data...</string>
   </property>
   <property name="shortcut">
    <string>Alt+E</string>
   </property>
  </action>
  <action name="actionVirtual_Keyboard">
   <property name="text">
    <string>Virtual Keyboard</string>
   </property>
   <property name="shortcut">
    <string>Alt+K</string>
   </property>
  </action>
  <action name="actionStep_Instruction">
   <property name="text">
    <string>Step Instruction</string>
   </property>
   <property name="shortcut">
    <string>Alt+I</string>
   </property>
  </action>
  <action name="actionCIA_Viewer">
   <property name="text">
    <string>CIA Viewer</string>
   </property>
  </action>
  <action name="actionRAM_Viewer">
   <property name="text">
    <string>RAM Viewer</string>
   </property>
   <property name="shortcut">
    <string>Alt+R</string>
   </property>
  </action>
  <action name="actionOpen_PRG">
   <property name="text">
    <string>Open PRG...</string>
   </property>
   <property name="shortcut">
    <string>Alt+O</string>
   </property>
  </action>
  <action name="actionBreakpoint_Editor">
   <property name="text">
    <string>Breakpoint Editor</string>
   </property>
   <property name="shortcut">
    <string>Alt+B</string>
   </property>
  </action>
  <action name="actionVIC_Viewer">
   <property name="text">
    <string>VIC Viewer</string>
   </property>
   <property name="shortcut">
    <string>Alt+V</string>
   </property>
  </action>
  <action name="actionStep_Line">
   <property name="text">
    <string>Step Line</string>
   </property>
   <property name="shortcut">
    <string>Alt+L</string>
   </property>
  </action>
  <action name="actionExtract_RAM_to_File">
   <property name="text">
    <string>Extract RAM to File...</string>
   </property>
  </action>
  <action name="actiondisable_joysticks">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>disabled</string>
   </property>
   <property name="shortcut">
    <string>Alt+0</string>
   </property>
  </action>
  <action name="actionjoystick_1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>joystick 1</string>
   </property>
   <property name="shortcut">
    <string>Alt+1</string>
   </property>
  </action>
  <action name="actionjoystick_2">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>joystick 2</string>
   </property>
   <property name="shortcut">
    <string>Alt+2</string>
   </property>
  </action>
  <action name="actionboth_joysticks">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>joystick 1+ 2</string>
   </property>
   <property name="shortcut">
    <string>Alt+3</string>
   </property>
  </action>
  <action name="actionVirtual_Joysticks">
   <property name="text">
    <string>Virtual Joysticks</string>
   </property>
   <property name="shortcut">
    <string>Alt+J</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
